homeassistant/components/tesla/sensor.py

```diff
@@ -87,6 +87,16 @@ def unit_of_measurement(self):
                 self._unit = LENGTH_KILOMETERS
                 self.current_value /= 0.621371
                 self.current_value = round(self.current_value, 2)
+        elif self.tesla_device.type == "charging rate sensor":
+            self.current_value = self.tesla_device.charging_rate
+            self._unit = self.tesla_device.measurement
+            self._attributes = {
+                "time_left": self.tesla_device.time_left,
```

Comment:
What kind of time is this and when is it updated? We normally don't allow relative times in the state machine since it leads to updates just due to time passing.

Reply:
This is the time in hours that the Tesla API returns for how long before charging will complete. This value is controlled by the vehicle and can change based on different conditions (amps, temperature, % complete) and is refreshed when the API is refreshed by the component (minimum of 5 minutes).

Reply:
Based on what you are saying, it the value type would be an `integer`, which is fine.  Essentially it's battery run time.  👍

Reply:
Sounds ok.